README.md

```diff
@@ -0,0 +1,171 @@
+Local test profile is set automatically through abstract integration test class. No need to worry about that.
+
+<a name="development-strategy"></a>
+## Development strategy
+The development strategy defines how the development is done for both backend and frontend applications and how the product is released to the end user. At first the focus should be making the minimum viable product, so that the foundation of the whole product is set and can be build upon.
+
+The step of develop is for this reason divided into releases - these releases define what is the end goal and what the product should look like after this release is done. Release should be defined before it's beginning and shouldn't be changed during the progression.
+
+<a name="release-1"></a>
+### Release 1
+
+<a name="release-2"></a>
+### Release 2
+_Not yet defined._
+
+<a name="contributing"></a>
+## Contributing
+
+<a name="code-standards"></a>
+### Code standards
+Java version 17 introduces a lot of new structures, features and enhancements, so the code should use them as much as it can to make the code as clean as possible using the most recent technology provided:
+- Type inference
+- Records
+- Stream api - including advanced methods
+- Improved switch statements
+
+Spring boot perspective:
+- Constructor injection instead of single field injection
+- Returning response entity in controllers
+- Having services implement their interfaces
+- Commented code which logs important information using SLF4J
+- Mappers without component model
+- Avoiding expression in mapping by using named methods
+
+<a name="changing-entity"></a>
+### Changing an entity
+Since the application uses migration, specifically liquibase, when changing an entity, it's required to also write a liquibase script. That can be done either by creating script manually, or running
+```shell
+mvn clean install -DskipTests liquibase:generateChangeLog
+```
+
+this will automatically generate diff in the `src/main/resources/liquibase` folder. This file has to be checked before getting onto git.
+New generated change log's name should start with 4 digit long unique string match this regex
+```regexp
+\d{4}-([a-zA-Z-])+
+```
+Also has to contain author's name and id of each change set has to match the name of the change log file with some unique number suffix, in regex words:
+```regexp
+\d{4}-([a-zA-Z-])+-\d+
+```
+
+<a name="testing"></a>
+### Testing
+Testing endpoints is possible through IntelliJ Idea http client, although it is not a recommended option. Better way to test endpoints is using [Postman](https://www.postman.com/)
+
+Each new logic should be tested manually through options mentioned above, but also tested using automatic tests - either unit or integration tests to ensure that each and every time something changes, pipeline tests all features and in case something fails, we'll know about it. Automatic tests should test most edge cases possible and ensure that the code works properly using these tests.
```

Comment:
```suggestion
Each new logic should be tested manually through options mentioned above, but also tested using automatic tests - either unit or integration tests to ensure that each and every time something changes, pipeline tests all features and in case something fails, we'll know about it. Automatic tests should test the most edge cases possible and ensure that the code works properly using these tests.
```